File: implementation/frost-scripts/performance-test/string-transform-workload.py
#!/usr/bin/env python3

n = 4000
words = list(map(lambda _: "AbCdEfGhIj", range(n)))
lowered = list(map(str.lower, words))
uppered = list(map(str.upper, lowered))

assert uppered[0] == "ABCDEFGHIJ"

result = sum(map(lambda s: 1 if s == "ABCDEFGHIJ" else 0, uppered))
assert result == n
print(result)

File: frost-scripts/performance-test/string-transform-workload.rb
#!/usr/bin/env ruby

n = 250_000
words = Array.new(n, "AbCdEfGhIj")
lowered = words.map(&:downcase)
uppered = lowered.map(&:upcase)

raise unless uppered[0] == "ABCDEFGHIJ"

result = uppered.count("ABCDEFGHIJ")
raise unless result == n
puts result
